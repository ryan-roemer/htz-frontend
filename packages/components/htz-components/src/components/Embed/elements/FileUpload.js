/* *************************************************************** *
 * This element accepts these inputTemplates:
  [
    com.polobase.fileUpload,
  ]
 * This element does not emits an onLoad event
 * *************************************************************** */

import React from 'react';
import PropTypes from 'prop-types';
import { FelaComponent, } from 'react-fela';
import HtzLink from '../../HtzLink/HtzLink';
import IconPdf from '../../Icon/icons/IconPdf';
import IconDrive from '../../Icon/icons/IconDrive';

// eslint-disable-next-line react/prop-types
const LinkToFile = ({ href, target, children, }) => (
  <FelaComponent
    style={({ theme, }) => ({
      color: theme.color('link'),
      textDecoration: 'none',

      ':hover': {
        color: theme.color('link'),
        borderBottom: '1px solid',
        paddingBottom: '.06em',
        outline: '0',
      },
    })}
  >
    {({ className, }) => (
      <HtzLink className={className} href={href} target={target}>
        {children}
      </HtzLink>
    )}
  </FelaComponent>
);

function FileUpload({ source, embedType, contentName, }) {
  const Icon = embedType === 'pdf' ? IconPdf : IconDrive;

  return (
    <LinkToFile href={source} target="_blank">
      <Icon size={5.5} />
      <FelaComponent style={{ marginStart: '1rem', }} as="span">
        {contentName}
      </FelaComponent>
    </LinkToFile>
  );
}

FileUpload.propTypes = {
  /**
   * Contains the file's URL.
   */
  source: PropTypes.string.isRequired,
  /**
   * File's type.
   */
  embedType: PropTypes.string.isRequired,
  /**
   * File's description.
   */
  contentName: PropTypes.string.isRequired,
};

export default FileUpload;
