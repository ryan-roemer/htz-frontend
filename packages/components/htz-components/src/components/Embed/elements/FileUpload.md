<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [Code Example](#FileUpload-component)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

### FileUpload component

An example for Uploaded file.

```jsx static
<FileUpload
  content = 'http://www.haaretz.co.il/st/inter/excel_upload/haaretz/2017/20170726-105102.xlsx'
  embedType = 'excel'
  contentName = 'some file to download'
/>
```
