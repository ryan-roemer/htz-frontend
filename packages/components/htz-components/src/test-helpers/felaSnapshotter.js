import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
