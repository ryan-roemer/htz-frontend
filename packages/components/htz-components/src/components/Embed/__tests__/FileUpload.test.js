import React from 'react';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';

import FileUpload from '../elements/FileUpload';

it('File upload ', () => {
  const snapshot = felaSnapshotter(
    <FileUpload
      content=""
      embedType="excel"
      elementType="embedElement"
      inputTemplate="com.polobase.fileUpload"
      contentId="7.4136"
      contentName="gdfjgh"
    />
  );
  expect(snapshot).toMatchSnapshot();
});
