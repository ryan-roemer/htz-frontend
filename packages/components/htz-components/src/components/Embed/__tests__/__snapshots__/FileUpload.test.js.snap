// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File upload  1`] = `
Object {
  "component": <div>
    <a
      className="htz-a htz-b htz-c htz-d htz-e"
      href=""
      target="_blank"
    >
      <i
        aria-hidden="true"
        className="htz-f htz-g htz-h htz-i htz-j htz-k htz-l htz-m htz-n htz-o htz-p htz-q htz-r htz-s htz-t htz-u htz-v htz-w"
        data-icon="-"
      />
      <span>
        gdfjgh
      </span>
    </a>
  </div>,
  "styles": ".htz-a{color:#0895c3}.htz-b{text-decoration:none}.htz-c:hover{border-bottom:1px solid}.htz-d:hover{padding-bottom:.06em}.htz-e:hover{outline:0rem}.htz-f{float:right}.htz-g{color:#e22134}.htz-h{background-color:transparent}.htz-i{border:0px}.htz-j{cursor:pointer}.htz-k{display:inline-block}.htz-l{padding:0rem}.htz-m{position:relative}.htz-n{text-align:center}.htz-o{vertical-align:middle}.htz-p{height:2.625rem}.htz-q{line-height:2.625rem}.htz-r{min-width:2.625rem}.htz-s{font-style:normal}.htz-t{width:1em}.htz-u{font-size:2.25rem}.htz-v{font-weight:700}.htz-w{text-rendering:optimizeLegibility}",
}
`;
